src/crawlee/parsel_crawler/_parsel_crawler.py

```diff
@@ -1,39 +1,30 @@
 from __future__ import annotations
 
-import asyncio
-import logging
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING
 
 from parsel import Selector
-from pydantic import ValidationError
 
-from crawlee import EnqueueStrategy
-from crawlee._request import BaseRequestData
-from crawlee._utils.blocked import RETRY_CSS_SELECTORS
 from crawlee._utils.docs import docs_group
-from crawlee._utils.urls import convert_to_absolute_url, is_url_absolute
-from crawlee.basic_crawler import BasicCrawler, BasicCrawlerOptions, ContextPipeline
-from crawlee.errors import SessionError
-from crawlee.http_clients import HttpxHttpClient
-from crawlee.http_crawler import HttpCrawlingContext
-from crawlee.parsel_crawler._parsel_crawling_context import ParselCrawlingContext
+from crawlee.parsel_crawler._parsel_parser import ParselParser
+from crawlee.static_content_crawler._static_content_crawler import StaticContentCrawler
+
+from ._parsel_crawling_context import ParselCrawlingContext
 
 if TYPE_CHECKING:
     from collections.abc import AsyncGenerator, Iterable
 
     from typing_extensions import Unpack
 
-    from crawlee._types import BasicCrawlingContext, EnqueueLinksKwargs
+    from crawlee.basic_crawler import BasicCrawlerOptions
+    from crawlee.static_content_crawler._static_crawling_context import ParsedHttpCrawlingContext
 
 
 @docs_group('Classes')
-class ParselCrawler(BasicCrawler[ParselCrawlingContext]):
+class ParselCrawler(StaticContentCrawler[ParselCrawlingContext, Selector]):
```

Comment:
The current constructor:

```python
    def __init__(
        self,
        *,
        additional_http_error_status_codes: Iterable[int] = (),
        ignore_http_error_status_codes: Iterable[int] = (),
        **kwargs: Unpack[BasicCrawlerOptions[ParselCrawlingContext]],
    ) -> None:
```    

Couldn't we move the `additional_http_error_status_codes` and `ignore_http_error_status_codes` one layer above? Same for other crawlers at this level. It seems they're just passing them to the parent.

And maybe we could also create a new typed dict for the options: `StaticContentCrawler(BasicCrawlerOptions)`

Does it make sense?

Reply:
Done